public/app/features/dashboard-scene/settings/variables/VariableEditorForm.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { FormEvent } from 'react';
+
+import { SelectableValue, VariableType } from '@grafana/data';
+import { selectors } from '@grafana/e2e-selectors';
+import { SceneVariable, AdHocFiltersVariable, AdHocFilterSet } from '@grafana/scenes';
+import { VariableHide, defaultVariableModel } from '@grafana/schema';
+import { HorizontalGroup, Button } from '@grafana/ui';
+import { ConfirmDeleteModal } from 'app/features/variables/editor/ConfirmDeleteModal';
+// import { VariableHideSelect } from 'app/features/variables/editor/VariableHideSelect';
+import { VariableHideSelect } from 'app/features/variables/editor/VariableHideSelect';
+import { VariableLegend } from 'app/features/variables/editor/VariableLegend';
+import { VariableTextAreaField } from 'app/features/variables/editor/VariableTextAreaField';
+import { VariableTextField } from 'app/features/variables/editor/VariableTextField';
+import { VariableNameConstraints } from 'app/features/variables/editor/types';
+
+import { VariableTypeSelect } from './components/VariableTypeSelect';
+import { getVariableEditor, getVariableScene, isEditableVariableType } from './utils';
+
+interface VariableEditorFormProps {
+  variable: SceneVariable;
+  onSubmit: (variable: SceneVariable) => void;
+}
+
+export function VariableEditorForm({ variable, onSubmit }: VariableEditorFormProps) {
+  const [editingVariable, setEditingVariable] = React.useState(variable.clone());
+
+  const { name, type, label, description, hide } = editingVariable.useState();
+  const EditorToRender = isEditableVariableType(type) ? getVariableEditor(type) : undefined;
+
+  const onTypeChange = (option: SelectableValue<VariableType>) => {
+    const variableType = option.value && isEditableVariableType(option.value) ? option.value : undefined;
+
+    if (!variableType) {
+      return;
+    }
+
+    const scene = getVariableScene(variableType);
+
+    // FIXME: Remove set and type as required properties from AdHocFiltersVariable in @grafana/scenes
+    if (scene === AdHocFiltersVariable) {
+      const newVariable = new scene({ name, description, set: new AdHocFilterSet({}), type: 'adhoc' });
+      setEditingVariable(newVariable);
+    } else {
+      // @ts-ignore TS complains about missing properties for AdHocFiltersVariable
+      const newVariable = new scene({ name, description });
+      setEditingVariable(newVariable);
+    }
+  };
+
```

Comment:
not sure this approach will work, as the variable being edited is not part of the dashboard scene (or SceneVariableSet), so you won't be able to run it / view options (values).

Reply:
Ups, you're right, we cannot resolve the dependencies if we clone it. Probably we need to update the current one.

@axelavargas and I were concerned about how frustrating can be, not to be able to restore the previous variable configuration without discarding all the dashboard changes.

This can be solved, by cloning the original state and adding a discard button to go back, but for the moment, I can leave the approach the same as we do for the links. The changes applies automatically when updating a field

Reply:
@ivanortegaalba - I've talked just briefly with @axelavargas about discarding the changes as a default, as an alternative to the Apply that we have today. I think this approach would make sense, but let's talk with @thanos-karachalios and @Ijin08 about the impact on the UX. 

Overall I think it should be ok to capture the variable state before going to variable edit, not clone it and manipulate the original object, but restore the captured state on discard.

Reply:
Applied @dprokop